pkg/azureconstants/azure_constants.go

```diff
@@ -216,6 +218,7 @@ const (
 	DefaultAzureClientAttachDetachRateLimiterQPS    = (240.0 / 180.0)                                          // Default compute QPS limit is 240 queries / 3 minutes
 	DefaultAzureClientAttachDetachRateLimiterBucket = int(DefaultAzureClientAttachDetachRateLimiterQPS * 60.0) // Allow for a burst of a minutes worth of quota
 	DefaultAzureClientAttachDetachBatchInitialDelay = 1 * time.Second                                          // Wait 1s before processing a batch of attach or detach disk requests
+	DefaultVolumeAttachRetryLimit                   = 2
```

Comment:
@sunpa93 Is 2 a large enough default?

Reply:
It occurs to me that we should have different retry limits based on what kind of errors we have retrieved. In case of attach failure due to `contextTimeout`, 2 retries would add up to 15 minutes of waiting for attach operation to complete as that would be total 3 attach attempts with 5 minutes timeout each. However, in case of attach failures due to `operationPreemption`, where the operation usually fails early, 2 retries may not give sufficient amount of time.

Reply:
I did two experiments of 500 pods scale test with the retry limit of **3**. Each of them has 40-50 times of retriable error, totally only one CRI retried twice while others just used one retry. I'll keep the retry limit to 2 as discussed and we can improve it in the future when we get more ideas about how it works.